docs/protocol/gas.md

```diff
@@ -151,13 +164,31 @@ You don't buy gas, instead, the gas fee is automatically removed from your accou
 
 The only exception to this rule is when you make a function call to a contract. In this case, you need to define how many gas units to use, up to a maximum value of `300Tgas`. This amount will be converted to $NEAR using the network's gas price and deducted from your account's balance.
 
-Since many transactions will take more [than 1 block to execute](./transaction-execution.md), and the gas price is recalculated on each block and could go up, you will be charged a pessimistic estimate of $NEAR (see details below).
+If the transaction ends up using less gas than the amount deducted, the difference will simply be **refunded to your account**. However, the network subtracts a **gas refund fee** of at leat 1 Tgas or up to 5% of the unspent gas.
+
+<details>
+
+<summary> Gas Refund Fee </summary>
+
+Since protocol version 78, the unspent gas at the end of receipt execution is subject to a gas refund fee. The fee is calculated as `max(1 Tgas, 0.05 * unspent_gas) * gas_price`. The gas price is from the time of purchase.
 
-If the transaction ends up using less gas than the amount deducted, the difference will simply be **refunded to your account**.
+But why introduce such a fee, when refunding 100% of unspent gas allows developers to just attach the maximum amount of gas and not worry about gas estimations too much?
+
+The resason is that attaching too much gas to function calls makes the network less efficient.
+
+Especially congestion control between shards is tricky when transactions have much more gas attached than they actually use. The network limits how many cross contract calls can target a single shard per chunk, to avoid huge queues of incoming receipts on a shard. The limit looks at attached gas as an upper boundary for how much work the receipt causes on the receiving shard. Attaching too much gas can cause this limit to be too restrictive, which stalls shards unnecessarily.
```

Comment:
```suggestion
Congestion control between shards gets tricky when transactions have much more gas attached than they actually use. The network limits how many cross contract calls can target a single shard per chunk, to avoid huge queues of incoming receipts on a shard. This limit sees the attached gas as an upper boundary for how much work the receipt causes on the receiving shard. Attaching too much gas can cause this limit to become too restrictive, which stalls shards unnecessarily.
```